module/Company/src/Company/Controller/AdminController.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+namespace Company\Controller;
+
+use Zend\Mvc\Controller\AbstractActionController;
+use Zend\View\Model\ViewModel;
+
+class AdminController extends AbstractActionController
+{
+
+    public function indexAction()
+    {
+        $companyService = $this->getCompanyService();
+        
+        $vm = new ViewModel(array(
+            'company_list' => $companyService->getHiddenCompanyList()
+        ));
+        
+        return $vm;
+
+    }
+    public function addCompanyAction()
+    {
+        $companyService = $this->getCompanyService();
+        $companyForm=$companyService->getCompanyForm();
+        $request = $this->getRequest();
+        if ($request->isPost()) {
+            if (!isset($companyName)){
+                $companyName=$request->getPost()['asciiName'];
+            }
+            $companyService = $this->getCompanyService();
+            $companyForm=$companyService->getCompanyForm();
+            $companyForm->setData($request->getPost());
+
+            // TODO: isValid does not work yet
+            if ($companyForm->isValid()) {
+                $company=$companyService->insertCompany();
+                $company->exchangeArray($request->getPost()); // Temporary fix, bind does not work yet?
+                $companyService->saveCompany();
+                return $this->redirect()->toRoute('admin_company/default', array('action'=>'edit', 'asciiCompanyName'=>$companyName),array(),false);   
+            }
+        }
+        //$company=$companyService->insertCompany();
+        //$companyForm->bind($company);
+        $companyForm->setAttribute('action',$this->url()->fromRoute('admin_company/default',array('action'=>'addCompany')));
+        $vm = new ViewModel(array(
+          //  'company' => $company,
+            'companyEditForm' => $companyForm,
+        ));
+        
+        return $vm;
+        
+    }
+    public function addJobAction()
+    {
+        $companyService = $this->getCompanyService();
+        $companyName = $this->params('asciiCompanyName');    
+        
+        $companyForm=$companyService->getJobForm();
+        
+        $request = $this->getRequest();
+        if ($request->isPost()) {
+            if (!isset($jobName)){
+                $jobName=$request->getPost()['asciiName'];
+            }
+            $companyService = $this->getCompanyService();
+            $companyForm=$companyService->getJobForm();
+            $companyForm->setData($request->getPost());
+
+            // TODO: isValid does not work yet
+            if ($companyForm->isValid()) {
+                $job=$companyService->insertJobForCompanyAsciiName($companyName);
+                $job->exchangeArray($request->getPost()); 
+                $companyService->saveCompany();
+                return $this->redirect()->toRoute('admin_company/editCompany/editJob', array('asciiCompanyName'=>$companyName, 'jobName'=>$jobName),array(),true);   
+            }
```

Comment:
Het is misschien een aanrader om dit allemaal even te herformateren volgens PSR1/2